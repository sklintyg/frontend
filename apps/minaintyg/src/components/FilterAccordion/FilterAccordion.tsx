import { IDSIconChevron } from '@frontend/ids-react-ts'
import type { ReactNode } from 'react'

export function FilterAccordion({ total, listed, noun, children }: { noun: string; total: number; listed: number; children: ReactNode }) {
  return (
    <details className="group mb-5">
      <summary className="flex cursor-pointer select-none flex-col md:flex-row">
        <div className="grow border-y border-stone-clear py-2.5 md:pl-5">
          Visar {listed} av {total} {noun}
        </div>
        <div className="-mt-px flex justify-between gap-5 border-y border-stone-clear py-2.5 md:mt-0 md:border-l md:px-5">
          Filtrera lista
          <IDSIconChevron width="100%" height="100%" className="h-2.5 w-2.5 rotate-90 self-center group-open:-rotate-90" />
        </div>
      </summary>
      <div className="border-b border-stone-clear py-5">{children}</div>
    </details>
  )
}
