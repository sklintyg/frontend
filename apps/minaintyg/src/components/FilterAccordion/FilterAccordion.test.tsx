import { render } from '@testing-library/react'
import { FilterAccordion } from './FilterAccordion'

it('Should render as expected', () => {
  const { container } = render(
    <FilterAccordion title="Headline">
      <span>Test</span>
    </FilterAccordion>
  )
  expect(container).toMatchSnapshot()
})
