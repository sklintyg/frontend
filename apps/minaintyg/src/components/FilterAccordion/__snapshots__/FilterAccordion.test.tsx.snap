// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render as expected 1`] = `
<div>
  <details
    class="group mb-5"
    open=""
  >
    <summary
      class="flex cursor-pointer select-none flex-col md:flex-row"
    >
      <div
        class="border-stone-clear grow border-y py-2.5 md:pl-5"
      >
        Headline
      </div>
      <div
        class="border-stone-clear -mt-px flex justify-between gap-5 border-y py-2.5 md:mt-0 md:border-l md:px-5"
      >
        Filtrera lista
        <ids-icon-chevron
          class="h-2.5 w-2.5 rotate-90 self-center group-open:-rotate-90 ids-icon"
        />
      </div>
    </summary>
    <div
      class="border-stone-clear border-b py-5"
    >
      <span>
        Test
      </span>
    </div>
  </details>
</div>
`;
